tests/Formatters/AbstractFormatterTests.cs

```diff
@@ -66,50 +67,67 @@ protected AbstractFormatterTest()
         {
-            TestCode = testCode;
+            var text = SourceText.From(testCode, encoding);
+            TestState.Sources.Add(text);
 
             var solution = GetSolution(TestState.Sources.ToArray(), TestState.AdditionalFiles.ToArray(), TestState.AdditionalReferences.ToArray());
             var project = solution.Projects.Single();
             var document = project.Documents.Single();
-            var options = (OptionSet)await document.GetOptionsAsync();
             var formatOptions = new FormatOptions(
                 workspaceFilePath: project.FilePath,
                 isSolution: false,
                 logLevel: LogLevel.Trace,
                 saveFormattedFiles: false,
                 changesAreErrors: false,
-                filesToFormat: ImmutableHashSet.Create<string>(document.FilePath));
-
-            ICodingConventionsSnapshot codingConventions = new TestCodingConventionsSnapshot(editorConfig);
-            options = OptionsApplier.ApplyConventions(options, codingConventions, Language);
+                filesToFormat: ImmutableHashSet.Create(document.FilePath));
 
-            var filesToFormat = new[] { (document, options, codingConventions) }.ToImmutableArray();
+            var filesToFormat = await GetOnlyFileToFormatAsync(solution, editorConfig);
 
             var formattedSolution = await Formatter.FormatAsync(solution, filesToFormat, formatOptions, Logger, default);
-            var formattedDocument = formattedSolution.Projects.Single().Documents.Single();
+            var formattedDocument = GetOnlyDocument(formattedSolution);
             var formattedText = await formattedDocument.GetTextAsync();
 
             Assert.Equal(expectedCode, formattedText.ToString());
+
+            return formattedText;
         }
 
+        /// <summary>
+        /// Gets the only <see cref="Document"/> along with related options and conventions.
+        /// </summary>
+        /// <param name="solution">A Solution containing a single Project containing a single Document.</param>
+        /// <param name="editorConfig">The editorconfig to apply to the documents options set.</param>
+        /// <returns>The document contained within along with option set and coding conventions.</returns>
+        protected async Task<ImmutableArray<(Document, OptionSet, ICodingConventionsSnapshot)>> GetOnlyFileToFormatAsync(Solution solution, IReadOnlyDictionary<string, string> editorConfig)
+        {
+            var document = GetOnlyDocument(solution);
+            var options = (OptionSet)await document.GetOptionsAsync();
+
+            ICodingConventionsSnapshot codingConventions = new TestCodingConventionsSnapshot(editorConfig);
+            options = OptionsApplier.ApplyConventions(options, codingConventions, Language);
+
+            return new[] { (document, options, codingConventions) }.ToImmutableArray();
```

Comment:
It looks like you were just migrating this existing code around, but please consider changing this to use `ImmutableArray.Create`